### PR TITLE
Test Cleanup 01

### PR DESCRIPTION
**Pull request by nathanwoctopusdeploy**
# Background

Prior to Async Halibut we are performing some test cleanup.

This PR got a bit big and includes:

Renaming abcTests to abcFixture
Move Fixtures to sub folders to mirror where the sut lives
Move Util/* to Support/* due to conflicts with a Fixture in Util
Move other files to Support that made sense
Change some tests to use ClientServiceBuilder
Change some tests to use an attribute to run for both Polling and Listening, removing duplicate tests or just added another test case where it made sense.

# How to review this PR

Quality :heavy_check_mark:

# Pre-requisites

- [ ] I have read [How we use GitHub Issues](https://github.com/OctopusDeploy/Issues/blob/master/docs/CONTRIBUTING.internal.md) for help deciding when and where it's appropriate to make an issue.
- [ ] I have considered informing or consulting the right people, according to the [ownership map](https://whimsical.com/ownership-map-NzbiD4HJyvhC9jNJNfS6TG).
- [ ] I have considered appropriate testing for my change.

### CODE REVIEW

File: source/Halibut.Tests/Support/TeamCityDetection.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+using System;
```
Comment (nathanwoctopusdeploy) [inline source/Halibut.Tests/Support/TeamCityDetection.cs:1]: This was just a move 🤷

**Comment by nathanwoctopusdeploy** [comment]
[sc-53136]

File: source/Halibut.Tests/Support/ClientServiceBuilder.cs [hunk 2]
```diff
@@ -78,19 +78,34 @@ public ClientServiceBuilder WithPortForwarding()
...
+        }
+
         public ClientAndService Build()
         {
             serviceFactory = serviceFactory ?? new DelegateServiceFactory();
 
-            var octopus = new HalibutRuntimeBuilder().WithServerCertificate(clientCertAndThumbprint.Certificate2)
-                .WithLogFactory(new TestContextLogFactory("Tentacle"))
-                .Build();
+            var octopusLogFactory = new TestContextLogFactory("Server");
```
Comment (LukeButters) [inline source/Halibut.Tests/Support/ClientServiceBuilder.cs:98]: Should this be `Client`?

File: source/Halibut.Tests/UsageFixture.cs [hunk 3]
```diff
@@ -35,37 +35,23 @@ public void OctopusCanDiscoverTentacle()
...
-            var services = GetDelegateServiceFactory();
-            services.Register<ISupportedServices>(() => new SupportedServices());
-            using (var clientAndService = ClientServiceBuilder.Polling().WithServiceFactory(services).Build())
-            {
-                var svc = clientAndService.CreateClient<ISupportedServices>();
-                for (var i = 1; i < 100; i++)
+                for (var i = 0; i < 2000; i++)
                 {
-                    var i1 = i;
-                    svc.GetLocation(new MapLocation { Latitude = -i, Longitude = i }).Should().Match<MapLocation>(x => x.Latitude == i1 && x.Longitude == -i1);
```
Comment (LukeButters) [inline source/Halibut.Tests/UsageFixture.cs:68]: Did we stop testing this call? Should we add another test that tests this call?

**Review by LukeButters — APPROVED** [review]
LGTM, consider adding back what looked like the dropped test

File: source/Halibut.Tests/UsageFixture.cs (hunk 3 above)
Comment (nathanwoctopusdeploy) [reply source/Halibut.Tests/UsageFixture.cs:68]: Does it add any value? Listening was doing one thing, Polling another. I made the Listening test do both!! We can add a 
listening and polling variations of this if we want!!

Comment (LukeButters) [reply source/Halibut.Tests/UsageFixture.cs:68]: I think we should, more tests cost very little.

File: source/Halibut.Tests/Support/ClientServiceBuilder.cs (hunk 2 above)
Comment (nathanwoctopusdeploy) [reply source/Halibut.Tests/Support/ClientServiceBuilder.cs]: Change 👍

File: source/Halibut.Tests/UsageFixture.cs (hunk 3 above)
Comment (nathanwoctopusdeploy) [reply source/Halibut.Tests/UsageFixture.cs:68]: Added the test back in with a polling and listening variation 👍

**Pull request merged**